sphinx_needs/directives/need.py

```diff
@@ -86,32 +86,29 @@ def run(self) -> Sequence[nodes.Node]:
         #############################################################################################
         env = self.env
 
-        delete_opt = self.options.get("delete")
-        if isinstance(delete_opt, str):
-            if delete_opt.upper() in ["TRUE", 1, "YES"]:
-                delete_opt = True
-            elif delete_opt.upper() in ["FALSE", 0, "NO"]:
-                delete_opt = False
-            else:
-                raise Exception("delete attribute must be true or false")
-
-        collapse = self.options.get("collapse")
-        if isinstance(collapse, str):
-            if collapse.upper() in ["TRUE", 1, "YES"]:
-                collapse = True
-            elif collapse.upper() in ["FALSE", 0, "NO"]:
-                collapse = False
-            else:
-                raise Exception("collapse attribute must be true or false")
-
-        jinja_content = self.options.get("jinja_content")
-        if isinstance(jinja_content, str):
-            if jinja_content.upper() in ["TRUE", 1, "YES"]:
-                jinja_content = True
-            elif jinja_content.upper() in ["FALSE", 0, "NO"]:
-                jinja_content = False
-            else:
-                raise Exception("jinja_content attribute must be true or false")
+        def _get_boolean(option_name: str) -> bool | None:
+            option_value = self.options.get(option_name)
+            if isinstance(option_value, str):
+                if option_value.upper() in ["", "TRUE", "YES"]:
+                    return True
+                if option_value.upper() in ["FALSE", "NO"]:
+                    return False
+                raise ValueError(
+                    f"{option_name!r} option must be true/false/yes/no, found: {option_value!r}"
+                )
+            return None
```

Comment:
is `none` correct here?
Shouldn't it be `True` in cases where no string was provided, but the option was set?

Reply:
no, if the option is set then the value will be the empty string, which is now converted to `True` above,
I just changed it to return the `option_value` otherwise, to be the same as previous, although this should never not be `None` at this point

Reply:
I removed the `true` from here: https://sphinx-needs--1188.org.readthedocs.build/en/1188/directives/need.html#collapse, and you can see it does work correctly